skyportal/handlers/api/instrument.py

```diff
@@ -1,13 +1,18 @@
+import functools
 from marshmallow.exceptions import ValidationError
+from sqlalchemy.orm.exc import NoResultFound
 from baselayer.app.access import permissions, auth_or_token
+from tornado.ioloop import IOLoop
+
 from ..base import BaseHandler
-from ...models import DBSession, Instrument, Telescope
+from ...models import DBSession, Instrument, InstrumentField, Telescope
 from ...enum_types import ALLOWED_BANDPASSES
+from ...utils.instrument import get_mocs
 
 
 class InstrumentHandler(BaseHandler):
     @permissions(['System admin'])
-    def post(self):
+    async def post(self):
```

Comment:
Why is this function async?

Reply:
This is required for having the asynchronous call to the fields or not?

Reply:
I don't think so, unless you use an `await` inside.

Reply:
Ok I see. Removed.